### ISSUE TITLE
Queries with 'LIKE' and % wild cards are not created properly

### ISSUE THREAD

**Issue by YonatanRan** [cakephp/cakephp#8473]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.1.7
* Platform and Target: Apache 2.2, Ubuntu 14.04, PHP 5.6, Mysql 5.5

### What you did
Created a query with conditions that include LIKE %value%
Example:
$this->Users->find()->where(['meta LIKE' => '%reload%']);

### Expected Behavior
Expected to get proper results for the query (when writing the query manually results are received.)

### Actual Behavior
No results are received.
Using the Debug ToolKit (ToolBar) I can see the query and it seems that the query is built wrong.
Example:
meta like '"%reload%"' 

Notice that there seem to be to many ' or " around the value.
This seems to only be an issue with wild cards, could not recreate otherwise

P.S. Remember, an issue is not the place to ask questions. You can use [Stack Overflow](http://stackoverflow.com/questions/tagged/cakephp)
for that or join the #cakephp channel on irc.freenode.net, where we will be more
than happy to help answer your questions.

Before you open an issue, please check if a similar issue already exists or has been closed before.

**Comment by markstory** [comment]
I thought we had tests covering this, I'll double check that though.

**Comment by YonatanRan** [comment]
Wholly crap! Didn't expect a response so fast!

I was about to start digging  in to see if I could find the issue...

Nice!

Yonatan Ran

**Comment by markstory** [comment]
We [have a few existing tests](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Database/QueryTest.php#L506-L522) so I'm not sure what is different in your situation.

**Comment by YonatanRan** [comment]
This seems to be a fringe case and most likely only related to me and
something I did.

I created a new installation of CakePHP 3 and for testing and I cannot
recreate the situation. But it still persists in the existing installation
where I first detected this issue.

I think this can be closed.
I apologize for the opening the issue with out properly testing it first...

Thanks for the very quick response.

Yonatan Ran

On Tue, Mar 15, 2016 at 10:05 PM, Mark Story <notifications@github.com>
wrote:

> We have a few existing tests
> <https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Database/QueryTest.php#L506-L522>
> so I'm not sure what is different in your situation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cakephp/cakephp/issues/8473#issuecomment-197112846>
>



-- 
*Yonatan Ran*

Mobile: +1 386-490-3629  |  Email: Yonatan@visualead.com
Web: www.visualead.com | www.Shijuema.com
Skype: yonatan.ran
